cmd/cvc-operator/controller/cstorvolumeclaim.go

```diff
@@ -643,7 +676,349 @@ func addPDBLabelOnCVC(
+//        replicas) if applicable then return same pdb name. If not applicable do
+//        following changes:
+//        1.1.1 Delete PDB if no other CVC is pointing to PDB.
+//    1.2 If current volume was not pointing to any PDB then do nothing.
+//    1.3 If current volume is HAVolume then check is there any PDB already
+//        existing among the current replica pools. If PDB exists then return
+//        that PDB name. If PDB doesn't exist then create new PDB and return newely
+//        created PDB name.
+// 2. Update CVC label to point it to newely PDB got from above step and also
+//    replicas pool information on status of CVC.
+func handlePostScalingProcess(cvc *apis.CStorVolumeClaim,
+	currentRPNames []string) error {
+	var err error
+	cvcCopy := cvc.DeepCopy()
+	cvc.Status.PoolInfo = []string{}
+	cvc.Status.PoolInfo = append(cvc.Status.PoolInfo, currentRPNames...)
+	pdbName, err := updatePDBForVolume(cvc)
+	if err != nil {
+		return errors.Wrapf(err,
+			"failed to handle PDB for scaled volume %s",
+			cvc.Name,
+		)
+	}
+	delete(cvc.Labels, string(apis.PodDisruptionBudgetKey))
+	if pdbName != "" {
+		cvc.Labels[string(apis.PodDisruptionBudgetKey)] = pdbName
+	}
+	newCVCObj, err := cvclaim.NewKubeclient().WithNamespace(cvc.Namespace).Update(cvc)
+	if err != nil {
+		// If error occured point it to old cvc object it self
+		cvc = cvcCopy
+		return errors.Wrapf(err,
+			"failed to update %s CVC status with scaledup replica pool names",
+			cvc.Name,
+		)
+	}
+	// Point cvc to updated cvc object
+	cvc = newCVCObj
+	return nil
+}
+
+// verifyAndUpdateScaleUpInfo does the following changes:
+// 1. Get list of new replica pool names by using CVC(spec and status)
+// 2. Verify status of ScalingUp Replica(by using CV object) based on the status
+//    does following changes:
+//    2.1: If scalingUp was completed then update PDB accordingly(only if it was
+//         HAVolume) and update the replica pool info on CVC(API calls).
+//    2.2: If scalingUp was going then return error saying scalingUp was in
+//      progress.
+func verifyAndUpdateScaleUpInfo(cvc *apis.CStorVolumeClaim, cvObj *apis.CStorVolume) error {
+	// scaledRPNames contains the new replica pool names where entier data was
+	// reconstructed from other replicas
+	scaledRPNames := []string{}
+	pvName := cvc.GetAnnotations()[volumeID]
+	desiredPoolNames := cvclaim.GetDesiredReplicaPoolNames(cvc)
+	newPoolNames := util.ListDiff(desiredPoolNames, cvc.Status.PoolInfo)
+	replicaPoolMap := map[string]bool{}
+
+	replicaPoolNames, err := cvr.GetVolumeReplicaPoolNames(pvName, cvc.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get current replica pool information")
+	}
+
+	for _, poolName := range replicaPoolNames {
+		replicaPoolMap[poolName] = true
+	}
+	for _, poolName := range newPoolNames {
+		if _, ok := replicaPoolMap[poolName]; ok {
+			scaledRPNames = append(scaledRPNames, poolName)
```

Comment:
just return error as 'in-progress' if !ok and update the status when all the required CVRs are required